src/common/common_helpers.hpp

```diff
@@ -43,24 +47,62 @@ bool performWithoutThrowing(Action&& action) noexcept
 #endif
 }
 
+/// Converts an `errno` positive integer to an error code.
+///
+/// `0` is converted to `nullopt`.
+/// Internally in use to convert raw integers (from DSDL types) to optional error codes.
+///
+inline sdk::ErrorCode errnoToErrorCode(const int errno_value) noexcept
+{
+    CETL_DEBUG_ASSERT(errno_value > 0, "Positive Posix error number is expected.");
+
+    return static_cast<sdk::ErrorCode>(errno_value);
```

Comment:
This needs to be either a switch on errno, or ErrorCode need to become a `struct`. See https://github.com/OpenCyphal-Garage/opencyphal-vehicle-system-management-daemon/issues/19#issuecomment-2721676870

Reply:
I thought we agreed that the first step would be `optional`.

Reply:
You mean the rest comes in a separate PR?

Reply:
yes